app/src/main/java/com/portfolio/tasky/entry/fragments/LoginFragment.kt

```diff
@@ -1,56 +1,137 @@
-import com.portfolio.tasky.FragmentInflaterImpl
+import com.portfolio.tasky.viewModels.LoginViewModel
+import com.portfolio.tasky.views.TaskyAppCompatEditText
+import dagger.hilt.android.AndroidEntryPoint
 
-class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl() {
+@AndroidEntryPoint
+class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl(), TextChanged  {
     private lateinit var viewBinding: LayoutLoginBinding
 
+    private val viewModel: LoginViewModel by viewModels()
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        viewBinding = LayoutLoginBinding.inflate(inflater, container, false)
+        viewBinding = DataBindingUtil.inflate(inflater, R.layout.layout_login, container, false)
+        viewBinding.lifecycleOwner = this
         return viewBinding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        setObservers()
         viewBinding.btnLogin.setOnClickListener {
             openRegistrationFragment()
         }
     }
 
+    private fun setObservers() {
+        viewModel.emailChange.observe(viewLifecycleOwner){
+            val emailField = viewBinding.etEmail
+            it?.let { it1 -> emailField.setValid(it1)
+                emailField.setError(emailField.subLayout.etInput.text?.isNotEmpty() as Boolean && !it1)}
+        }
+        viewModel.passwordChange.observe(viewLifecycleOwner){
+            val passwordField = viewBinding.etPassword
+            it?.let { it1 -> passwordField.setValid(it1)
+                passwordField.setError(passwordField.subLayout.etInput.text?.isNotEmpty() as Boolean && !it1)}
+        }
+        viewModel.areFieldsValid.observe(viewLifecycleOwner){
+            it.let {  viewBinding.btnLogin.isEnabled = it == true }
+        }
+    }
+
+    private fun setFieldValidations(loginFragment: LoginFragment?) {
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        val etPassword = viewBinding.etPassword.subLayout.etInput
+
+        etEmail.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail,
+                it
+            )
+        })
+        etPassword.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etPassword,
+                it
+            )
+        })
+    }
+
+    override fun onTextChanged(editable: Editable, taskyAppcompatEditText: TaskyAppCompatEditText) {
+        if (taskyAppcompatEditText.id == R.id.et_email){
+            viewModel.emailChange(editable.toString() as CharSequence)
+        }
+
+
+        if(taskyAppcompatEditText.id == R.id.et_password){
+            viewModel.passwordChange(editable.toString() as CharSequence)
+        }
+            viewModel.areFieldsValid()
+    }
```

Comment:
If you do what I tell you in the email comment, then this method and the whole interface can be deleted ;)